content/post/kubestash-v2024.12.9/index.md

```diff
@@ -0,0 +1,195 @@
+```yaml
+apiVersion: core.kubestash.com/v1alpha1
+kind: BackupVerifier
+metadata:
+  name: mysql-query-verifier
+  namespace: demo
+spec:
+  function: kubedbverifier
+  restoreOption:
+    target:
+      apiGroup: kubedb.com
+      kind: MySQL
+      name: sample-mysql
+      namespace: verify
+    addonInfo:
+      name: mysql-addon
+      tasks:
+        - name: logical-backup-restore
+  scheduler:
+    schedule: "*/5 * * * *"
+    jobTemplate:
+      backoffLimit: 1
+  type: Query
+  query:
+    mySQL:
+      - database: auth
+        table: user
+        rowCount:
+          operator: Equal
+          value: 100
+      - database: store
+        table: order
+  sessionHistoryLimit: 2
+```
+
+Now we need to update `BackupConfiguration` to refer the `BackupVerifier`. 
+
+```yaml
+apiVersion: core.kubestash.com/v1alpha1
+kind: BackupConfiguration
+metadata:
+  name: sample-backup
+  namespace: demo
+spec:
+  target:
+    apiGroup: kubedb.com
+    kind: MySQL
+    name: sample-mysql
+    namespace: demo
+  backends:
+  - name: gcs-storage
+    storageRef:
+      namespace: demo
+      name: gcs-storage
+    retentionPolicy:
+      name: demo-retention
+      namespace: demo
+  sessions:
+  - name: freq-backup
+    sessionHistoryLimit: 3
+    scheduler:
+      schedule: "*/5 * * * *"
+      jobTemplate:
+        backoffLimit: 1
+    repositories:
+    - name: demo-storage
+      backend: gcs-storage
+      backupVerifier: # updated backupVerifier
+        name: mysql-query-verifier
+        namespace: demo
+      directory: /dep/mysql
+      encryptionSecret:
+        name: encryption-secret
+        namespace: demo
+    addon:
+      name: mysql-addon
+      tasks:
+      - name: logical-backup
+    retryConfig:
+      maxRetry: 2
+      delay: 1m
+```
+
+Here, `.spec.sessions[*].repositories[*].backupVerifier` contains the `BackupVerifier` reference. 
+
+#### Templating for Exec Hook Commands
+
+We've added support for [Go template](https://pkg.go.dev/text/template) in exec type as well. Previously it was only limited to httpPost hook. Here is an example of configuring template in exec hook:
+
+```yaml
+apiVersion: core.kubestash.com/v1alpha1
+kind: HookTemplate
+metadata:
+  name: sample-hook
+  namespace: demo
+spec:
+  usagePolicy:
+    allowedNamespaces:
+      from: All
+  action:
+    exec:
+      command:
+      - curl
+      - -X
+      - POST
+      - -H
+      - "Content-Type: application/json"
+      - -d
+      - `{"text":":x: Name: {{ .Name }} Namespace: {{.Namespace}} Phase: {{.Status.Phase}}"}`
+      - https://slack-webhook.url/kubestash
+  executor:
+    type: Pod
+    pod:
+      selector: name=test-app, test=hook
+```
+
+
+### Improvements & Bug Fixes
+
+- We've fixed an issue for external databases backup. The database version was not resolved during backup which is fixed within this release.
+- We've fixed an issue where addon job template was not resolved in the `BackupConfiguration`. That has been fixed in this release.
+
```

Comment:
- We’ve fixed an issue with external database backups where the database version was not resolved during the backup process. This has now been fixed in this release.
- We have also fixed an issue where the addon job template specified in the BackupConfiguration was not applied properly. It has been fixed in this release.